### PR TITLE
[MM-48900] Display available recording files in call thread root post

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

I found a couple of issues while implementing this:

1. Multiple recording files were not saved correctly in post props due to a type mismatch.
2. There was a potential race condition in a rapid sequence of stop recording -> start recording that would lead to an invalid state with the host and users thinking a seeing the new recording as ended while ongoing.
3. Fixed a potential state parsing bug due to improper key length check.

As far as UX changes so far I only implemented a simple indicator of available recordings, no extra behaviour as there are some inconsistencies with the design so anything more will require further input.

#### Screenshot

image

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-48900

### CODE REVIEW

File: server/session.go [hunk 1]
```diff
@@ -124,6 +124,7 @@ func (p *Plugin) addUserSession(userID, connID string, channel *model.Channel) (
 		if userID == botID {
 			if state.Call.Recording != nil && state.Call.Recording.StartAt == 0 {
 				state.Call.Recording.StartAt = time.Now().UnixMilli()
+				state.Call.Recording.BotConnID = connID
```
Comment (cpoile) [inline server/session.go:127]: I like the change, but I'm not sure I totally understand what the race condition was.

**Review by cpoile — APPROVED** [review]
Nice! Just one question for my own understanding.

File: server/session.go (hunk 1 above)
Comment (streamer45) [reply server/session.go:127]: @cpoile Apologies, let me try to explain it properly with a triggering sequence:

With a recording already ongoing:

1. Host stops recording
2. Recording state is saved as ended and websocket event sent out.
3. Client receives ws event and updates local state, recording has ended.
4. Host quickly starts a new recording before recording bot from previous recording leaves call.
6. Recording bot for previous recording job eventually leaves the call, improperly setting the current recording as ended.

Comment (cpoile) [reply server/session.go:127]: Were you able to reproduce this? If so, does the second recording start? (considering we're using the same channel state)

Comment (streamer45) [reply server/session.go:127]: Yes it's pretty easy to reproduce. Recording starts but the client gets the stop event right away and then the state is tainted so both client and server think there's no recording on-going while the job is actually running and recording until the call ends.

Comment (cpoile) [reply server/session.go:127]: Cool :)

**Comment by streamer45** [comment]
@tanmay-des Heads up that we are going to merge this soon mostly because we have some important fixes in it and would like to prepare a dot release. Anything outstanding UX wise will be taken in a subsequent PR.

**Review by tanmay-des — APPROVED** [review]
Sounds good!

**Pull request merged**